gcloud/datastore/_implicit_environ.py

```diff
@@ -79,6 +86,16 @@ def compute_engine_id():
         connection.close()
 
 
+def _get_production_dataset_id():
```

Comment:
I didn't put much though into this name, any ideas for a replacement? The reason for "production" was that `_DATASET_ENV_VAR_NAME` (i.e. `'GCLOUD_DATASET_ID'`) is what we want production apps to use.

As for having the one-liner, the whole point of this PR is better tests, not better code. The one-liner seemed like the best way to separate `_determine_default_dataset_id` from the behavior of the 4 actions in said method.